build/assets/js/form_validation.js

```diff
@@ -5,93 +5,60 @@
 /* https://css-tricks.com/form-validation-part-2-constraint-validation-api-javascript/
 /*---------------------------------------*/
 
-//Name Validate
-var nameInput = {
-  target : document.getElementById('your-name'),
-  func : function(){
-    if (nameInput.target.validity.valueMissing){
-      nameInput.target.setCustomValidity('Please fill out this field.');
-    } else {
-      nameInput.target.setCustomValidity('');
-    }
-  },
-  init: function(){
-    this.target.addEventListener('input', this.func, false);
-    this.target.addEventListener('invalid', this.func, false);
-    this.target.addEventListener('change',this.func, false);
+class Input {
```

Comment:
今回は同一ファイル内にしていますが、Inputクラスは別ファイルにしたい。

Reply:
```suggestion
class Input {
```
クラスだけのファイルを作ることが運用上望ましいということですか？

Reply:
そうですね！ Inputクラスはこちらのページのフォームのみで利用するのではなく、他のページでも利用できるよう汎用的なものにしています。
イメージ感としてはinput.jsを新しく作成し、他のページでも利用できるInputクラスと汎用的なメソッド(function) を記載します。
form_validation.js にて、input.jsをimportして、このページに依存する処理(ビジネスロジック)を記載するというのが理想です！

Reply:
なるほど、importするのですね。汎用性的に使えるようファイルの分け方も今後意識していきます。